### PR TITLE
TB25: feat(version) - index versioning

### PR DESCRIPTION
**Pull request by thanh-nguyen-dang**
### New Features
- Implemented Versioning for the Index

### Deployment changes
- You may need to change the config map to use the new mapping file if the commons still use the old version of mapping file.

### CODE REVIEW

File: tube/spark/es_writer.py [hunk 1]
```diff
@@ -1,54 +1,178 @@
...
+def get_backup_index_name(index, version):
+    return '{}_{}'.format(version, index)
+
+
+def get_backup_alias(index):
+    return '{}_backup'.format(index)
+
+
+def get_backup_version(index_name):
+    res = re.match('.?([0-9]+)*', index_name)
```
Comment (m0nhawk) [inline tube/spark/es_writer.py]: Why the regex is so complex? Won't it result in the same as: `.?([0-9]*)`?

Comment (thanh-nguyen-dang) [reply tube/spark/es_writer.py]: need to have + because we need the number appear at least once

Comment (m0nhawk) [reply tube/spark/es_writer.py]: Then `.?([0-9]+)`? I doesn't fully understand the need of both `+` for digits and `*` for the whole group. There can be multiple groups?

Comment (thanh-nguyen-dang) [reply tube/spark/es_writer.py]: * is used to point that the whole group appear at the beginning of the string and we don't care the remaining of the string

**Pull request merged**